apis/v1alpha1/cluster_bpf_application_types.go

```diff
@@ -75,76 +74,219 @@ const (
+	// seen by the network device. When using the TCX program type, the tcx field
+	// is required. See tcx for more details on TCX programs.
+	//
+	// When set to Tracepoint, the program can attach to one of the predefined set
+	// of Linux kernel functions. When using the Tracepoint program type, the
+	// tracepoint field is required. See tracepoint for more details on Tracepoint
+	// programs.
+	//
+	// When set to UProbe, the program can attach in user-space. The UProbe is
+	// attached to a binary, library or function name, and optionally an offset in
+	// the code. When using the UProbe program type, the uprobe field is required.
+	// See uprobe for more details on UProbe programs.
+	//
+	// When set to URetProbe, the program can attach in user-space.
+	// The URetProbe is attached to the return of a binary, library or function
+	// name, and optionally an offset in the code.  When using the URetProbe
+	// program type, the uretprobe field is required. See uretprobe for more
+	// details on URetProbe programs.
+	//
+	// When set to XDP, the eBPF program can attach to network devices (interfaces)
+	// and will be called on every incoming packet received by the network device.
+	// When using the XDP program type, the xdp field is required. See xdp for more
+	// details on XDP programs.
+	//
 	// +unionDiscriminator
 	// +required
 	// +kubebuilder:validation:Enum:="XDP";"TC";"TCX";"FEntry";"FExit";"KProbe";"KRetProbe";"UProbe";"URetProbe";"TracePoint"
 	Type EBPFProgType `json:"type"`
 
-	// xdp defines the desired state of the application's XdpPrograms.
+	// xdp is an optional field, but required when the type field is set to XDP.
+	// xdp defines the desired state of the application's XDP programs. XDP program
+	// can be attached to network devices (interfaces) and will be called on every
+	// incoming packet received by the network device. The XDP hook point is just
+	// after the packet has been received off the wire, but before the Linux kernel
+	// has allocated an sk_buff, which is used to pass the packet through the
+	// kernel networking stack.
 	// +unionMember
 	// +optional
 	XDP *ClXdpProgramInfo `json:"xdp,omitempty"`
 
-	// tc defines the desired state of the application's TcPrograms.
+	// tc is an optional field, but required when the type field is set to TC. tc
+	// defines the desired state of the application's TC programs. TC programs are
+	// attached to network devices (interfaces). The program can be attached on
+	// either packet ingress or egress, so the program will be called on every
+	// incoming or outgoing packet seen by the network device. The TC hook point is
+	// in Linux's Traffic Control (tc) subsystem, which is after the Linux kernel
+	// has allocated an sk_buff. TCX is newer implementation of TC with enhanced
+	// performance and better support for running multiple programs on a given
+	// network device. This makes TC useful for packet classification actions.
```

Comment:
I suggest you delete the blurb about TCX here.  It's stated below in the TCX section.